In [ ]:
!pip install fastai
from fastai.text.all import *

In [ ]:
path = untar_data(URLs.IMDB)
dls = TextDataLoaders.from_folder(path=path,valid='test')
dls.show_batch()

With the function untar_data we decompress and store the Windows path of the dataset in the variable path.
URLs is a module that contains paths that point to datasets widely used in research and academia. In this case we use a dataset from IMDB that contains movie reviews


After we load our data into a DataLoader that will use the validation set as test set.
Batch is a small set of the data that will be fed to the model and we output it using the show_batch method

In [ ]:
learn = text_classifier_learner(dls,AWD_LSTM,metrics= accuracy)
learn.fine_tune(4,1e-2)

We use a pretrained model called AWD_LSTM that has initialised weights so using fit_one_cycle is pointless so we use fine_tune method with 4 epochs and a learning rate of 0.01.Data is stored in the dls object and to assess our model we use accuracy which is correctly predicted data divided by the number of all instances in the test set

In [ ]:
learn.predict('It was  good')

('pos', tensor(1), tensor([0.1287, 0.8713]))

In this last stage we use our model for inference. We write a comment as a string and we see if it is going to classify it as positive or negative. In this case the review: It was good was classified as positive with probability 0.8713